cli/command_snapshot_fix.go

```diff
@@ -0,0 +1,188 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/snapshot"
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+)
+
+type commandSnapshotFix struct {
+	removeObjectIDs    []string
+	removeFilesByName  []string
+	fullRead           bool
+	removeInvalidFiles bool
+
+	// snapshots to target
+	manifestIDs []string
+	sources     []string
+
+	commit bool
+
+	verifier *snapshotfs.Verifier
+}
+
+func (c *commandSnapshotFix) setup(svc appServices, parent commandParent) {
+	cmd := parent.Command("fix", "Fix snapshot consistency issues")
+
+	// action to perform
+	cmd.Flag("remove-invalid-files", "Delete invalid files").BoolVar(&c.removeInvalidFiles)
+	cmd.Flag("remove-object", "Remove objects with specified ID").StringsVar(&c.removeObjectIDs)
+	cmd.Flag("remove-file", "Remove file with specified name (unqualified)").StringsVar(&c.removeFilesByName)
+
+	// scope of rewrite
+	cmd.Flag("manifest-id", "Manifest IDs").StringsVar(&c.manifestIDs)
+	cmd.Flag("source", "Source to target (username@hostname:/path)").StringsVar(&c.sources)
+
+	// verification options
+	cmd.Flag("full-read", "Perform full read when verifying files").BoolVar(&c.fullRead)
+
+	cmd.Flag("commit", "Update snapshot manifests").BoolVar(&c.commit)
```

Comment:
Forgive my ignorance: what is the point of having this default to off?

Reply:
Safety, mostly. If the `fix` ends up deleting the entire repository (because of some broad deletion rule), that's not really helpful.